### PR TITLE
Update ConfigDump documentation.

### PR DESCRIPTION
**Pull request by paul-r-gall**
Signed-off-by: Paul Gallagher <pgal@google.com>

Commit Message: update ConfigDump documentation
Additional Description: ScopedRoutesConfigDump and SecretsConfigDump were missing.
Risk Level: Low
Testing: NA
Docs Changes: Yes
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: api/envoy/admin/v3/config_dump.proto [hunk 1]
```diff
@@ -57,7 +57,9 @@ message ConfigDump {
   // * *clusters*: :ref:`ClustersConfigDump <envoy_v3_api_msg_admin.v3.ClustersConfigDump>`
   // * *endpoints*:  :ref:`EndpointsConfigDump <envoy_v3_api_msg_admin.v3.EndpointsConfigDump>`
   // * *listeners*: :ref:`ListenersConfigDump <envoy_v3_api_msg_admin.v3.ListenersConfigDump>`
+  // * *scoped_routes*: :ref:`ScopedRoutesConfigDump <envoy_v3_api_msg_admin.v3.ScopedRoutesConfigDump>`
   // * *routes*:  :ref:`RoutesConfigDump <envoy_v3_api_msg_admin.v3.RoutesConfigDump>`
+  // * *secrets*:  :ref:`SecretsConfigDump <envoy_v3_api_msg_admin.v3.SecretsConfigDump>`
```
Comment (antoniovicente) [inline api/envoy/admin/v3/config_dump.proto:62]: Do you happen to know what PRs added support for these additional types?

Comment (paul-r-gall) [reply api/envoy/admin/v3/config_dump.proto:62]: History search indicates that ScopedRouteConfigDump was added in https://github.com/envoyproxy/envoy/pull/6932, and SecretsConfigDump was added in https://github.com/envoyproxy/envoy/pull/7365.

**Comment by antoniovicente** [comment]
CI found some issues with generation of shadow protos from the proto you edited.  Could you run `./tools/proto_format/proto_format.sh fix` and push a version that contains the changes made by that script.

/wait

**Review by lizan — APPROVED** [review]

**Comment by antoniovicente** [comment]
Please merge upstream/main.  The QUIC coverage failure is deterministic so merging main is needed to fix the CI failure.

**Review by antoniovicente — APPROVED** [review]

**Pull request merged**